Pkl/PklTypes/DataSize.cs

```diff
@@ -0,0 +1,14 @@
+namespace Pkl.PklTypes;
+
+public class DataSize
```

Comment:
Wondering if this is a good use case for a struct rather than a class, same goes for the Duration class

Reply:
Yeah I agree, using structs for these types would make sense. Other than those two I'll make `IntSeq` a struct too.